File: app/src/main/java/torille/fi/lurkforreddit/subreddit/SubredditComponent.java
package torille.fi.lurkforreddit.subreddit;

import dagger.Component;
import torille.fi.lurkforreddit.data.RedditRepositoryComponent;
import torille.fi.lurkforreddit.di.scope.FragmentScoped;

/**
 * Created by eva on 24.4.2017.
 */
@FragmentScoped
@Component(dependencies = RedditRepositoryComponent.class, modules = {SubredditPresenterModule.class})
public interface SubredditComponent {

    void inject(SubredditFragment fragment);
}

File: app/src/main/java/torille/fi/lurkforreddit/subreddit/SubredditComponent.kt
package torille.fi.lurkforreddit.subreddit

import dagger.Component
import torille.fi.lurkforreddit.data.RedditRepositoryComponent
import torille.fi.lurkforreddit.di.scope.FragmentScoped

@FragmentScoped
@Component(dependencies = arrayOf(RedditRepositoryComponent::class), modules = arrayOf(SubredditPresenterModule::class))
interface SubredditComponent {

    fun inject(fragment: SubredditFragment)

}
